backend/apps/moderation/serializers.py

```
@@ -0,0 +1,42 @@
+            "content_type_str", "content_summary", "content_type_app", "content_type_model",
+            "category", "description", "status", "action_taken", "created_at"
+        ]
+        read_only_fields = ["reporter", "status", "action_taken", "created_at", "content_type"]
+
+    def create(self, validated_data):
+        app = validated_data.pop("content_type_app")
+        model = validated_data.pop("content_type_model")
+        try:
+            ct = ContentType.objects.get(app_label=app, model=model)
+        except ContentType.DoesNotExist:
+            raise serializers.ValidationError({"content_type": "Invalid app or model"})
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)